third-party/rspamd/src/test/java/org/apache/james/rspamd/client/RSpamDHttpClientTest.java

```diff
@@ -0,0 +1,172 @@
+
+import io.restassured.http.Header;
+import io.restassured.specification.RequestSpecification;
+
+public class RSpamDHttpClientTest {
+    private final static String SPAM_MESSAGE_PATH = "mail/spam/spam1.eml";
+    private final static String HAM_MESSAGE_PATH = "mail/ham/ham1.eml";
+
+    @RegisterExtension
+    static DockerRSpamDExtension rSpamDExtension = new DockerRSpamDExtension();
+
+    private MimeMessage spamMessage;
+    private MimeMessage hamMessage;
+
+    @BeforeEach
+    void setup() throws MessagingException {
+        spamMessage = MimeMessageUtil.mimeMessageFromStream(ClassLoader.getSystemResourceAsStream(SPAM_MESSAGE_PATH));
+        hamMessage = MimeMessageUtil.mimeMessageFromStream(ClassLoader.getSystemResourceAsStream(HAM_MESSAGE_PATH));
+    }
+
+    @Test
+    void checkMailWithWrongPasswordShouldThrowUnauthorizedExceptionException() throws IOException {
+        RSpamDClientConfiguration configuration = new RSpamDClientConfiguration(rSpamDExtension.getBaseUrl(), "wrongPassword", Optional.empty());
+        RSpamDHttpClient client = new RSpamDHttpClient(configuration);
+
+        assertThatThrownBy(() -> client.checkV2(spamMessage).block())
+            .hasMessage("{\"error\":\"Unauthorized\"}")
+            .isInstanceOf(UnauthorizedException.class);
+    }
+
+    @Test
+    void learnSpamWithWrongPasswordShouldThrowUnauthorizedExceptionException() throws IOException {
+        RSpamDClientConfiguration configuration = new RSpamDClientConfiguration(rSpamDExtension.getBaseUrl(), "wrongPassword", Optional.empty());
+        RSpamDHttpClient client = new RSpamDHttpClient(configuration);
+
+        assertThatThrownBy(() -> reportAsSpam(client, spamMessage))
+            .hasMessage("{\"error\":\"Unauthorized\"}")
+            .isInstanceOf(UnauthorizedException.class);
+    }
+
+    @Test
+    void learnHamWithWrongPasswordShouldThrowUnauthorizedExceptionException() throws IOException {
+        RSpamDClientConfiguration configuration = new RSpamDClientConfiguration(rSpamDExtension.getBaseUrl(), "wrongPassword", Optional.empty());
+        RSpamDHttpClient client = new RSpamDHttpClient(configuration);
+
+        assertThatThrownBy(() -> reportAsHam(client, hamMessage))
+            .hasMessage("{\"error\":\"Unauthorized\"}")
+            .isInstanceOf(UnauthorizedException.class);
+    }
+
+    @Test
+    void checkSpamMailUsingRSpamDClientWithExactPasswordShouldReturnAnalysisResultAsSameAsUsingRawClient() throws IOException, MessagingException {
+        RSpamDClientConfiguration configuration = new RSpamDClientConfiguration(rSpamDExtension.getBaseUrl(), PASSWORD, Optional.empty());
+        RSpamDHttpClient client = new RSpamDHttpClient(configuration);
+
+        AnalysisResult analysisResult = client.checkV2(spamMessage).block();
+        assertThat(analysisResult).isEqualTo(AnalysisResult.builder()
+            .action(AnalysisResult.Action.REJECT)
+            .score(16.1F)
+            .requiredScore(15.0F)
```

Comment:
Checkv2 spam1 manually through Rspamd web interface, just exactly as the test suite...

![image](https://user-images.githubusercontent.com/55171818/180745632-bfeb27c0-7bbe-42a2-a6d1-13e049e75bc5.png)

Reply:
weird, the score is not the same with different machines